src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,188 @@
+
+# Requirement 26: The srs_id value in a gpkg_geometry_columns table row
+# SHALL be an srs_id column value from the gpkg_spatial_ref_sys table.
+#
+# Requirement 27: The z value in a gpkg_geometry_columns table row SHALL be one
+# of 0, 1, or 2.
+#
+# Requirement 28: The m value in a gpkg_geometry_columns table row SHALL be one
+# of 0, 1, or 2.
+#
+# Requirement 146: The srs_id value in a gpkg_geometry_columns table row
+# SHALL match the srs_id column value from the corresponding row in the
+# gpkg_contents table.
+function gpkgtable(db, ; layer=1)
+  rowtable = DBInterface.execute(
+    # According to https://www.geopackage.org/spec/#r16 
+    # Values of the gpkg_contents table srs_id column 
+    # SHALL reference values in the gpkg_spatial_ref_sys table srs_id column
+    # According to https://www.geopackage.org/spec/#r18
+    # The gpkg_contents table SHALL contain a row 
+    # with a lowercase data_type column value of "features" 
+    # for each vector features user data table or view.
+    db,
+    """
+    SELECT g.table_name AS tablename, g.column_name AS columnname, 
+    c.srs_id AS srsid, g.z, srs.organization AS org, srs.organization_coordsys_id AS orgcoordsysid,
+    ( SELECT type FROM sqlite_master WHERE lower(name) = lower(c.table_name) AND type IN ('table', 'view')) AS object_type
+    FROM gpkg_geometry_columns g, gpkg_spatial_ref_sys srs
+    JOIN gpkg_contents c ON ( g.table_name = c.table_name )
+    WHERE c.data_type = 'features'
+    AND object_type IS NOT NULL
+    AND g.srs_id = srs.srs_id
+    AND g.srs_id = c.srs_id
+    AND g.z IN (0, 1, 2)
+    AND g.m = 0
+    LIMIT $layer;
+    """
+  )
+
+  # Note: first feature table that is read specifies the CRS to be used on all feature tables resulted from SELECT statement
+  firstrow = first(rowtable)
+
+  # According to https://www.geopackage.org/spec/#r33, feature table geometry columns
+  # SHALL contain geometries with the srs_id specified 
+  # for the column by the gpkg_geometry_columns table srs_id column value.
+  srsid, org, orgcoordsysid = firstrow.srsid, firstrow.org, firstrow.orgcoordsysid
+
+  # an srs_id of 0 uses undefined Geographic CRS
+  if iszero(srsid)
+    crs = LatLon{WGS84Latest}
+    # if srs_id not equal to -1
+  elseif srsid != -1
+    # CRS assigned by the org 
+    if org == "EPSG"
+      # orgcoordsysid is the numeric id of the CRS assigned by the org
+      crs = CoordRefSystems.get(EPSG{orgcoordsysid})
+    elseif org == "ESRI"
+      crs = CoordRefSystems.get(ERSI{orgcoordsysid})
+    end
+  else
+    # defaults to undefined Cartesian CRS
+    crs = Cartesian{NoDatum}
+  end
+
+  results = map((row.tablename, row.columnname) for row in rowtable) do (tablename, columnname)
+    # According to https://www.geopackage.org/spec/#r14
+    # The table_name column value in a gpkg_contents table row 
+    # SHALL contain the name of a SQLite table or view.
+    tableinfo = SQLite.tableinfo(db, tablename)
```

Comment:
Here we are making a direct call to `SQLite.tableinfo`. Do we have an equivalent in DBInterface.jl to avoid mixing the two interfaces in the implementation?